boards/shields/sparkfun_sara_r4/boards/disco_l475_iot1.conf

```diff
@@ -0,0 +1 @@
+CONFIG_UART_4=y
```

Comment:
@mike-scott , why not applying @MaureenHelm's proposal (in https://github.com/zephyrproject-rtos/zephyr/pull/14057/commits/d5eda49fe83f7e53c8af7926eafdda025786c5fe#r263920854) here?

This is close to what we should get in future, if get read of Kconfig symbols per device instance. Then, the following would do the same:
```
&arduino_spi {
               status = "ok";
}
```

Reply:
@erwango This PR is now suffering from "scope creep".  There are several more additions I want to do, but felt like it would be best to land it as is and then make this changes.

Reply:
Also, this approach doesn't work for WNC-M14A2A (the original modem driver in this PR) -- it uses non-standard pins for UART on the Arduino headers.

Reply:
> @erwango This PR is now suffering from "scope creep". 

I'd rather say this is an ambitious PR that could have been split in small pieces.
Please not that I retained myself to request again for splitting board connector dts definitions into dedicated .dtsi files :-).

Anyway, agreed to let this as a further enhancement.